src/Illuminate/Auth/SessionGuard.php

```diff
@@ -210,6 +210,69 @@ protected function userFromRecaller($recaller)
         return $user;
     }
 
+    /**
+     * Impersonate the given user.
+     *
+     * @param  \Illuminate\Contracts\Auth\Authenticatable  $user
+     * @return void
+     *
+     * @throws \Illuminate\Auth\AuthenticationException
+     */
+    public function impersonate(AuthenticatableContract $user)
+    {
+        if (! $authenticated = $this->user()) {
+            throw new AuthenticationException('Cannot impersonate without a currently authenticated user.');
```

Comment:
The wording seems off here.. hmm, some alternatives?

> Cannot impersonate without an being authenticated.

> Cannot impersonate without already being authenticated.

> Impersonation requires you to be an authenticated user .

Reply:
Yeah I'm not sure -- I'll wait for additional feedback on this 🙏

Reply:
also can you use translation?

Reply:
@lloricode No exception messages are translated in Laravel. They are not visible to end users.